### PR TITLE
cclient: implement iota_api_broadcast_transactions

### LINKED ISSUE #43: Implement broadcast_transactions

### PR DESCRIPTION
**Pull request by furuame**
Implement the API `broadcastTransactions`, close #43 

# Test Plan:
CI must pass

### CODE REVIEW

File: cclient/request/broadcast_transactions.c [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#include "broadcast_transactions.h"
```
Comment (oopsmonk) [inline cclient/request/broadcast_transactions.c:8]: please add the full standard copyright. see `add_neighbors.c`

File: cclient/serialization/tests/test_json.c [hunk 2]
```diff
@@ -481,7 +481,98 @@ void test_deserialize_attach_to_tangle(void) {
...
-  // TODO
+  serializer_t serializer;
+  const char* json_text =
+      "{\"command\":\"broadcastTransactions\",\"trytes\":["
+      "\"BYSWEAUTWXHXZ9YBZISEK9LUHWGMHXCGEVNZHRLUWQFCUSDXZHOFHWHL9MQPVJXXZLIXPX"
+      "PXF9KYEREFSKCPKYIIKPZVLHUTDFQKKVVBBN9ATTLPCNPJDWDEVIYYLGPZGCWXOBDXMLJC9V"
+      "O9QXTTBLAXTTBFUAROYEGQIVB9MJWJKXJMCUPTWAUGFZBTZCSJVRBGMYXTVBDDS9MYUJCPZ9"
+      "YDWWQNIPUAIJXXSNLKUBSCOIJPCLEFPOXFJREXQCUVUMKSDOVQGGHRNILCO9GNCLWFM9APMN"
+      "MWYASHXQAYBEXF9QRIHIBHYEJOYHRQJAOKAQ9AJJFQ9WEIWIJOTZATIBOXQLBMIJU9PCGBLV"
+      "DDVFP9CFFSXTDUXMEGOOFXWRTLFGV9XXMYWEMGQEEEDBTIJ9OJOXFAPFQXCDAXOUDMLVYRMR"
```
Comment (tsvisabo) [inline cclient/serialization/tests/test_json.c]: can you relocate this string to an external definition/variable?

File: cclient/serialization/tests/test_json.c [hunk 3]
```diff
@@ -481,7 +481,98 @@ void test_deserialize_attach_to_tangle(void) {
...
+      "SXQWZCHDQSLGK9UOHCFKBIBNETK999999999999999999999999999999999999999999999"
+      "999999999999999999999999999999999999NOXDXXKUDWLOFJLIPQIBRBMGDYCPGDNLQOLQ"
+      "S99EQYKBIU9VHCJVIPFUYCQDNY9APGEVYLCENJIOBLWNB999999999XKBRHUD99C99999999"
+      "NKZKEKWLDKMJCI9N9XQOLWEPAYWSH9999999999999999999999999KDDTGZLIPBNZKMLTOL"
+      "OXQVNGLASESDQVPTXALEKRMIOHQLUHD9ELQDBQETS9QFGTYOYWLNTSKKMVJAUXSIROUICDOX"
+      "KSYZTDPEDKOQENTJOWJONDEWROCEJIEWFWLUAACVSJFTMCHHXJBJRKAAPUDXXVXFWP9X9999"
+      "IROUICDOXKSYZTDPEDKOQENTJOWJONDEWROCEJIEWFWLUAACVSJFTMCHHXJBJRKAAPUDXXVX"
+      "FWP9X9999");
+
+  serializer.vtable.broadcast_transactions_serialize_request(&serializer, req,
```
Comment (tsvisabo) [inline cclient/serialization/tests/test_json.c]: can we test it via: `iota_api_broadcast_transactions` instead?

**Review by tsvisabo — CHANGES_REQUESTED** [review]

File: cclient/serialization/tests/test_json.c (hunk 2 above)
Comment (furuame) [reply cclient/serialization/tests/test_json.c]: Good idea. But only for this API?

File: cclient/serialization/tests/test_json.c (hunk 3 above)
Comment (furuame) [reply cclient/serialization/tests/test_json.c]: Other iota_api_xxx should also do it. I think this should be treated another issue.

File: cclient/request/broadcast_transactions.c (hunk 1 above)
Comment (furuame) [reply cclient/request/broadcast_transactions.c:8]: Okay.

File: cclient/serialization/tests/test_json.c (hunk 2 above)
Comment (tsvisabo) [reply cclient/serialization/tests/test_json.c]: for any api that might need this :)

File: cclient/serialization/tests/test_json.c (hunk 3 above)
Comment (tsvisabo) [reply cclient/serialization/tests/test_json.c]: yes

**Review by tsvisabo — CHANGES_REQUESTED** [review]

**Review by thibault-martinez — APPROVED** [review]

File: cclient/request/broadcast_transactions.c (hunk 1 above)
Comment (thibault-martinez) [reply cclient/request/broadcast_transactions.c]: Copyright still missing

**Review by tsvisabo — APPROVED** [review]

**Pull request merged**